packages/react-select/src/components/Select/useSelect.ts

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { getPartitionedNativeProps, resolveShorthand } from '@fluentui/react-utilities';
+import type { SelectProps, SelectSlots, SelectState } from './Select.types';
+
+/**
+ * Array of all shorthand properties listed as the keys of SelectSlots
+ */
+export const selectShorthandProps: (keyof SelectSlots)[] = ['select', 'icon', 'root'];
+
+/**
+ * Create the state required to render Select.
+ *
+ * The returned state can be modified with hooks such as useSelectStyles,
+ * before being passed to renderSelect.
+ *
+ * @param props - props from this instance of Select
+ * @param ref - reference to the `<select>` element in Select
+ */
+export const useSelect = (props: SelectProps, ref: React.Ref<HTMLSelectElement>): SelectState => {
+  const { select, icon, root, size = 'medium', appearance = 'outline', inline } = props;
+
+  const nativeProps = getPartitionedNativeProps({
+    props,
+    primarySlotTagName: 'select',
+    excludedPropNames: ['appearance', 'inline', 'size'],
```

Comment:
```suggestion
    excludedPropNames: ['size'],
```

Are `appearance` and `inline` native attributes ? I can't seem to find any reference documentation about them